app/screens/select_server/select_server.test.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import SelectServer from './select_server.js';
+import {shallowWithIntl} from 'test/intl-test-helper';
+
+describe('SelectServer', () => {
+    const actions = {
+        getPing: jest.fn(),
+        handleServerUrlChanged: jest.fn(),
+        scheduleExpiredNotification: jest.fn(),
+        loadConfigAndLicense: jest.fn(),
+        login: jest.fn(),
+        resetPing: jest.fn(),
+        setLastUpgradeCheck: jest.fn(),
+        setServerVersion: jest.fn(),
+    };
+
+    const baseProps = {
+        actions,
+        hasConfigAndLicense: true,
+        serverUrl: '',
+    };
+
+    test('should match snapshot empty URL string', async () => {
+        const wrapper = shallowWithIntl(
+            <SelectServer {...baseProps}/>,
+        );
+        wrapper.instance().handleConnect();
+        expect(wrapper.state().error).not.toBeNull();
+        expect(wrapper.state().error.intl.id).toMatch('mobile.server_url.empty');
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
```

Comment:
I'd suggest to test this using [@testing-library/react-native](https://github.com/callstack/react-native-testing-library) so that test is slightly represented in user perspective instead of code implementation. With that, it can be something like:
```javascript
const {getByTestId, getByText} = renderWithReduxIntl(
    <SelectServer {...baseProps}/>,
);

const button = getByText('Connect');
fireEvent.press(button);

expect(getByTestId('error_text')).toBeTruthy();
expect(getByText('Please enter a valid server URL')).toBeTruthy();
```

If you do, please add the ff on top of the file.
```javascript
import {renderWithReduxIntl} from 'test/testing_library';
import {fireEvent, waitFor} from '@testing-library/react-native';

jest.mock('react-native-navigation', () => ({
    Navigation: {
        events: jest.fn(() => ({
            bindComponent: jest.fn(() => {
                return {remove: jest.fn()};
            }),
        })),
    },
}));
```